delta/subcommands/classify.py

```diff
@@ -135,16 +144,94 @@ def classify_image(model, image, label, path, net_name, options):
     if label:
         cm = predictor.confusion_matrix()
         class_names = list(map(lambda x: x.name, config.dataset.classes))
-        print_classes(cm)
         if len(config.dataset.classes) != cm.shape[0]:
             class_names = list(map(lambda x: 'Class %d' % (x), range(cm.shape[0])))
-        if options.confusion:
+        if options.confusion and (not shapes):
             save_confusion(cm, class_names,
                            os.path.join(out_path, 'confusion_' + os.path.splitext(base_out)[0] + '.pdf'))
         return cm
     return None
 
-def main(options):
+def get_wkt_path(image_path, wkt_folder=None):
+    '''Return the path to where the WKT file for an image should be'''
+    WKT_EXTENSION = '.wkt.csv'
+    if wkt_folder:
+        p = os.path.join(wkt_folder, os.path.basename(image_path))
+        path = os.path.splitext(p)[0] + WKT_EXTENSION
+    else:
+        path = os.path.splitext(image_path)[0] + WKT_EXTENSION
+    return path
+
+def load_shapes_matching_tag(wkt_path, tag):
+    '''Returns a list of all the shapes defined for this tag in the WKT file.
+       If tag is None, return untagged regions'''
+    shapes = []
+    with open(wkt_path, 'r') as f:
+        reader = csv.reader(f)
+        for row in reader:
+            raw_line = ', '.join(row)
+            if ('POLYGON' not in raw_line) or (len(row) != 2):
+                continue
+            names = row[1].strip().split(',')
+            if not tag and (names == ['']): # Match non-tagged regions
+                s = wkt.loads(row[0])
+                shapes.append(s)
+            else:
+                for n in names: # Check each tag of the region
+                    if tag == n.strip():
+                        s = wkt.loads(row[0])
+                        shapes.append(s)
+                        continue
+    return shapes
+
+def get_roi_containing_shapes(shapes) -> Rectangle:
+    '''Return a Rectangle containing all the shapes or None if none were passed in'''
+    if not shapes:
+        return None
+    roi = Rectangle(*shapes[0].bounds)
+    for i in range(1,len(shapes)):
+        new_roi = Rectangle(*shapes[i].bounds)
+        roi.expand_to_contain_rect(new_roi)
+    # Convert to integer values
+    return Rectangle(int(math.floor(roi.min_x)),
+                     int(math.floor(roi.min_y)),
+                     int(math.ceil(roi.max_x)),
+                     int(math.ceil(roi.max_y)))
+
+def shapes_to_pixel_coordinates(shapes, image_path):
```

Comment:
maybe some of these functions would be usable in the future and could be put in a shared file? maybe something in imagery package?

Reply:
I think that is a good idea once all of the classify features are done being added.